### PR TITLE
enhance transfer transaction form ux

### LINKED ISSUE #1562: Transfer Form. Improve mosaic amount input
- Display mosaic balance in the amount text box
- Display an error message when the transfer amount exceeds the balance

image

### PR DESCRIPTION
**Pull request by bassemmagdy**
-fixes #1562 
- rebasing #1701

### CODE REVIEW

File: src/components/AmountInput/AmountInputTs.ts [hunk 1]
```diff
@@ -36,12 +36,19 @@ import { NetworkType } from 'symbol-sdk';
...
             mosaics: 'mosaic/mosaics',
             networkType: 'network/networkType',
+            balanceMosaics: 'mosaic/balanceMosaics',
         }),
     },
 })
 export class AmountInputTs extends Vue {
     @Prop({ default: '' }) value: string;
     @Prop({ default: '' }) mosaicHex: string;
+    @Prop({ default: false }) isOffline: boolean;
```
[the remark this answers is not in the document]
Comment (bassemmagdy) [reply src/components/AmountInput/AmountInputTs.ts:46]: Added the missing checker to `AmountInput.vue`

File: src/components/AmountInput/AmountInput.vue [hunk 2]
```diff
@@ -9,13 +9,28 @@
         class="inputs-container"
     >
         <ErrorTooltip :errors="errors">
-            <input v-model="relativeValue" class="input-style input-size" type="text" />
+            <input v-model="relativeValue" class="input-style amount-input-size" type="text" />
+            <div v-if="!isOffline">
+                <input v-model="totalAvailableAmount" class="input-style total-amount-input-size" :disabled="true" />
+                <p v-if="isAmountGreaterThanBalance" style="color: red;">{{ $t('not_enough_balance') }}</p>
```
Comment (yilmazbahadir) [inline src/components/AmountInput/AmountInput.vue:15]: avoid inline styling as much as possible?

File: src/components/AmountInput/AmountInput.vue [hunk 3]
```diff
@@ -9,13 +9,28 @@
...
+.amount-input-size {
+    width: 60%;
+    height: 100%;
+}
+.total-amount-input-size {
+    width: 40%;
+    height: 100%;
+    color: #515a6e;
+    padding-left: 0.3rem;
+}
```
Comment (yilmazbahadir) [inline src/components/AmountInput/AmountInput.vue:35]: Should we move the styles to `AmountInput.less` to preserve consistency across components and use as below?
```
@import './AmountInput.less';
```

**Review by yilmazbahadir — APPROVED** [review]

**Comment by bassemmagdy** [comment]
updated in #1714

**Pull request closed without merging**